sros2/sros2/verb/generate_artifacts.py

```diff
@@ -44,8 +48,43 @@ def add_arguments(self, parser, cli_name):
 
     def main(self, *, args):
         try:
-            success = generate_artifacts(
+            success = _generate_artifacts(
                 args.keystore_root_path, args.security_contexts, args.policy_files)
         except FileNotFoundError as e:
             raise RuntimeError(str(e))
         return 0 if success else 1
+
+
+def _generate_artifacts(keystore_path=None, identity_names=[], policy_files=[]):
+    if keystore_path is None:
+        keystore_path = _get_keystore_path_from_env()
+        if keystore_path is None:
+            return False
+    if not _keystore.is_valid_keystore(keystore_path):
+        print('%s is not a valid keystore, creating new keystore' % keystore_path)
+        _keystore.create_keystore(keystore_path)
+
+    # create keys for all provided identities
+    for identity in identity_names:
+        if not _key.create_key(keystore_path, identity):
+            return False
+    for policy_file in policy_files:
+        policy_tree = load_policy(policy_file)
+        contexts_element = policy_tree.find('contexts')
+        for context in contexts_element:
+            identity_name = context.get('path')
+            if identity_name not in identity_names:
+                if not _key.create_key(keystore_path, identity_name):
+                    return False
+            policy_element = _policy.get_policy_from_tree(identity_name, policy_tree)
+            _permission.create_permissions_from_policy_element(
+                keystore_path, identity_name, policy_element)
+    return True
+
+
+def _get_keystore_path_from_env():
```

Comment:
this would be more suited in  `_utilities` and called from this file. This could come in handy for other verbs (e.g. if we make them use kwargs as suggested by @ruffsl)

Reply:
Alright, done. Refactored ever so slightly to pull the variable out into a global like the others in utility.